### ISSUE TITLE
Implement AOT safe DispatchProxy.Create

### ISSUE THREAD

**Issue by MichalStrehovsky** [dotnet/runtime#73136]
The generic DispatchProxy.Create can be done ahead of time, but it requires an AOT compiler feature.

**Comment by jkotas** [comment]
I do not think it make sense to have this custom build into Native AOT compiler like it was done in .NET Native. We should be looking at this as "reverse Invoke" reflection feature for all runtimes.

**Comment by steveharter** [comment]
> I do not think it make sense to have this custom build into Native AOT compiler like it was done in .NET Native. We should be looking at this as "reverse Invoke" reflection feature for all runtimes.

Is there an existing proposal for this?

@MichalStrehovsky do you plan on working on this - it was originally marked 8.0 so I assume it was important at that time at least.

cc @buyaa-n per our offline discussion

**Comment by MichalStrehovsky** [comment]
Not planning to work on it. There is an implementation in .NET Native that one can take inspiration from. There might be design docs as well but it was long time ago and I didn't work on any aspect of it.

**Comment by steveharter** [comment]
Moving to 8.0 for now but we need to discuss whether this is an AOT issue or a reflection\DispatchProxy issue.

**Comment by steveharter** [comment]
Moving to future; doesn't seem to be high enough priority to make the cut.

**Comment by steveharter** [comment]
We should take another look at this for v9.

**Comment by DjArt** [comment]
Is there any workaround?

**Comment by MichalStrehovsky** [comment]
> Is there any workaround?

No workaround, the implementation is full of Reflection.Emit and that's unsupportable in current form.

**Comment by mconnew** [comment]
Lack of support for DispatchProxy will block apps that use WCF from using the new UWP on .NET 9 capability as that uses native aot. Blog post announcing UWP on .NET 9: https://devblogs.microsoft.com/ifdef-windows/preview-uwp-support-for-dotnet-9-native-aot/
